core/src/main/java/jenkins/model/item_category/ItemCategory.java

```diff
@@ -0,0 +1,90 @@
+package jenkins.model.item_category;
+
+import hudson.Extension;
+import hudson.ExtensionPoint;
+import hudson.model.ModelObject;
+import jenkins.model.Messages;
+
+/**
+ * A category for {@link hudson.model.Item}s.
+ */
+public abstract class ItemCategory implements ModelObject, ExtensionPoint {
+
+    public static int MIN_WEIGHT = Integer.MIN_VALUE;
+
+    public static int MIN_TOSHOW = 1;
+
+    /**
+     * Identifier, e.g. "category-id-default", etc.
+     *
+     * @return the identifier
+     */
+    public abstract String getId();
+
+    /**
+     * The icon class specification e.g. 'category-icon-folder', 'category-icon-default', etc.
+     *
+     * @return the icon class specification
+     */
+    public abstract String getIconClassName();
+
+    /**
+     * The description in plain text
+     *
+     * @return the description
+     */
+    public abstract String getDescription();
+
+    /**
+     * Helpful to set the order.
+     *
+     * @return the weight
+     */
+    public abstract int getWeight();
+
+    /**
+     * Minimum number required to show the category.
+     *
+     * @return the minimum items required
+     */
+    public abstract int getMinToShow();
```

Comment:
I like Alternative-1. The complexity of the second 2 alternatives seems like a huge overkill. For 'Alternative-3' to make sense, I might as well keep the entire category definition in that UICategory interface, because what else is this interface supposed to do besides help render a UI? This already is the UICategory interface. If other consumers need a category interface and find this one too polluted, let them write their own. I don't expect either side of that condition to hold true, this interface won't be too polluted, and there won't be a bunch of other consumers. ...so why not let this interface include the optional parameters this consumer needs?